database-couchdb/index.js

```diff
@@ -1,103 +1,100 @@
+        }
+      },
+      'fields': ['_rev'],
+      'limit': 1,
+    }
+  )
+    .then((data) => {
+      return data.data.docs[0]._rev;
+    })
     .catch((err) => {
-      return err;
+      console.error(err);
     });
-};
+}
 
-const createNewData = (url, db) => {
-  let curId = startingItemId;
-  let availabilityStream = fs.createReadStream(availabilityPath);
-
-  const writeData = async (curIds, stores) => {
-    let count = 0;
-    let docs = [];
-    let entry = {};
-    let curId = curIds
-    availabilityStream
-      .on('error', (err) => {
-        console.error(err);
-      })
-      .pipe(csvParser())
-      .on('data', (data) => {
-
-        count++;
-        if (entry.itemId === undefined) {
-          entry.itemId = entry._id = curId.toString();
-          entry.itemAvailability = stores;
-        }
+exports.getData = (itemId) => {
 
-        entry.itemAvailability[parseInt(data.store_id) - 1].availability = data.availability;
-        if (count === 5) {
-          docs.push(entry);
-          entry = {};
-          curId++;
-          count = 0;
+  return axios.post(url + '/_find',
+    {
+      'selector': {
+        _id: {
+          '$eq': itemId
         }
-        if (docs.length === batchSize) {
-
-          console.log(curId);
-          availabilityStream.pause();
-
-          Axios.post(url + db + '/_bulk_docs', { docs })
-            .then(() => {
-              console.log('posted!');
-              setTimeout(() => {
-                console.log('Now data will start flowing again. ' + curId);
-
-                availabilityStream.resume();
-              }, 500);
-            })
-            .catch((err) => {
-              console.error(err);
-            });
-          docs = [];
-        }
-      })
-      .on('end', () => {
-        console.log('seed done!');
-        availabilityStream.destroy();
-      });
-  }
-  getStoreData((stores) => {
-    writeData(curId, stores);
-  });
+      },
+      'fields': ['itemId', 'itemAvailability'],
+      'limit': 1,
+    }
+  )
+    .then((data) => {
+      return data.data.docs[0];
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+};
 
-}
+exports.updateData = async (itemId, info) => {
 
-const getStoreData = (CB) => {
-  let storesStream = fs.createReadStream(storesPath);
-  let stores = [];
-  storesStream
-    .on('error', (err) => {
-      console.error(err);
-    })
-    .pipe(csvParser())
-    .on('data', (data) => {
-      let store = data;
-      store.storeId = data.store_id;
-      store.availability = false;
-      // console.log(store);
-      stores.push(store);
+  info._rev = await getRevId(itemId);
+
+  return axios.put(url + `/${itemId}`, info)
+    .then((data) => {
+      return data.data;
     })
-    .on('end', () => {
-      storesStream.destroy();
-      CB(stores);
+    .catch((err) => {
+      console.error(err);
     });
+};
+
+exports.deleteData = async (itemId) => {
 
+  let rev = await getRevId(itemId);
```

Comment:
This looks very easy to read, and I know exactly what's going on, good job writing such readable easy to follow code!